TELCODOCS-1092: GPU sharing methods (#67322)
openshift/openshift-docs: modules/nvidia-gpu-sharing-methods.adoc

// Module included in the following assemblies:
//
// * architecture/nvidia-gpu-architecture-overview.adoc

:_content-type: CONCEPT
[id="nvidia-gpu-sharing-methods_{context}"]
= GPU sharing methods

> With the rise of customers running GPU-accelerated computing on OpenShift for different workloads, Red Hat and NVIDIA have developed a GPU Operator to simplify GPU-accelerated computing on an enterprise-level container platform.

Reviewer: ```suggestion
With more customers running GPU-accelerated computing on OpenShift for different workloads, Red Hat and NVIDIA have developed a GPU Operator to simplify GPU-accelerated computing on an enterprise-level container platform.
```

Author: Made this change.

> Different applications have different compute requirements that could leave some of your GPUs underutilized. Providing the right amount of compute resources for each workload becomes critical to reduce deployment cost and maximize your GPU utilization.

Author: Changed.

> There are a number of different concurrency mechanisms for improving GPU utilization that range from programming model APIs to system software and hardware partitioning including virtualization. These GPU concurrency mechanisms include:

Author: Made these changes.

> * CUDA streams

Author: Changed to _* Compute Unified Device Architecture (CUDA) streams_

* Time-slicing

> * CUDA Multi-Process Service

Author: This is an NVIDIA product and they use caps. I did change to _* CUDA Multi-Process Service (MPS)_

* Multi-instance GPU (MIG)
* Virtualization with vGPU

> The following use cases provide GPU sharing suggestions for different scenarios:

Author: Made this change.

> * OpenShift bare metal - vGPU is not available. Consider using MIG-enabled cards.

Author: Made this change.

> * Openshift in VMs - vGPU is likely the best choice.

Author: Made this change.

> * Openshift with older nvidia cards with no MIG on bare metal - consider using time-slicing.

Author: Made this change.

> * Openshift in VMs with multiple GPUs and you want pass-through and vGPU - consider using separate VMs.

Author: Made this change.

> * Openshift bare metal with Openshift virualization and multiple GPUs - consider using pass-through for hosted VMs and time-slicing for containers.

Author: Changed previously. See above.
